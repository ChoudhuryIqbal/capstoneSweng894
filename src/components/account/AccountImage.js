import React from 'react'
import { Image } from 'react-native'

/**
 * Account Image componemt for Account forms
 * 
 * @param {*} props 
 */
const AccountImage = (props) => {
    const ImageHolder = () => {
        console.log(props.imageHolder);
        
        if(!props.imageHolder){
            return (
                <Image
                    source={props.placeholder}
                />
            )
        } else {
            return (
                <Image
                    source={props.image}
                />
            )
        }

    }
    return (
        <ImageHolder />
    )
}

export default AccountImage