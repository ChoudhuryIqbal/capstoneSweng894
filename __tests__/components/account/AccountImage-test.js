import React from 'react'
import AccountImage from '../../../src/components/account/AccountImage'
import {create} from 'react-test-renderer';

describe('Account Image correctly renders', () => {
    test('Matches the snapshot', () => {
        const accountImage = create(
            <AccountImage
            imageHolder={false}
            placeholder={require('../../../src/image/Placeholder150.png')}
        />
        )
        expect(accountImage.toJSON()).toMatchSnapshot();
    })
})
