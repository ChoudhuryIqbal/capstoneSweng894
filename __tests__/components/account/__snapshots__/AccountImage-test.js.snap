// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Image correctly renders Matches the snapshot 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/image/Placeholder150.png",
    }
  }
/>
`;
